src/main/java/org/opensearch/search/relevance/transformer/personalizeintelligentranking/reranker/impl/AmazonPersonalizedRankerImpl.java

```diff
@@ -82,14 +92,72 @@ public SearchHits rerank(SearchHits hits, PersonalizeRequestParameters requestPa
                     .withUserId(userId);
             GetPersonalizedRankingResult result = personalizeClient.getPersonalizedRanking(personalizeRequest);
 
-            //TODO: Combine Personalize and open search result. Change the result after transform logic is implemented
-            return hits;
+            List<PredictedItem> personalizeRrankingResult = result.getPersonalizedRanking();
+            // Build a map with key as item id and value as personalize ranking score
+            for (int j = 0; j < personalizeRrankingResult.size(); ++j) {
+                idToPersonalizeRankingScoreMap.put(personalizeRrankingResult.get(j).getItemId(),
+                        personalizeRrankingResult.get(j).getScore().floatValue());
+            }
+            // Build a map with key as item id and value as open search scores and another map
+            // with key as item id and value as corresponding search hit
+            for (int j = 0; j < numberOfHitsToRerank; ++j) {
+                idToOpenSearchScoreMap.put(originalHits.get(j).getId(), originalHits.get(j).getScore());
+                itemIdToSearchHitMap.put(originalHits.get(j).getId(), originalHits.get(j));
+            }
+
+            float weight = (float)rankerConfig.getWeight();
+            SearchHits newHits = combine_scores(idToPersonalizeRankingScoreMap, idToOpenSearchScoreMap,
+                    itemIdToSearchHitMap, hits.getTotalHits(), weight);
+
+            return newHits;
         } catch (Exception ex) {
             logger.error("Failed to re rank with Personalize. Returning original search results without Personalize re ranking.", ex);
             return hits;
         }
     }
 
+    //Combine open search hits and personalize campaign response
+    public SearchHits combine_scores(Map<String, Float> idToPersonalizeRankingScoreMap, Map<String, Float> idToOpenSearchScoreMap,
+                                     Map<String, SearchHit> itemIdToSearchHitMap, TotalHits totalHits, float w) {
```

Comment:
```suggestion
    public SearchHits combine_scores(Map<String, Float> idToPersonalizeRankingScoreMap, 
                                     Map<String, Float> idToOpenSearchScoreMap,
                                     Map<String, SearchHit> itemIdToSearchHitMap, 
                                     TotalHits totalHits, float weight) {
```

Reply:
The method naming here doesn't follow Java coding conventions. The method should be `combineScores`.